### ISSUE TITLE
Selecting deleted signal may crash the whole program

### ISSUE THREAD

**Issue by HLCaptain** [godotengine/godot#56163]
### Godot version

3.4.2.stable.official[45eaa2daf]

### System information

Windows 11 Pro N 22000.376

### Issue description

TLDR: {
When trying to Move to Method in a node's signal methods on the right tab, while the connected method is removed from the script which it is connected to (or the whole script is removed), the whole engine crashes. Sometimes it is "just an out of bounds error" written onto the standard output.
![image](https://user-images.githubusercontent.com/22623259/147116002-81677690-bc3b-47a4-8958-6903a1f34507.png)
Image of errors, when trying to Move to Method while the signal is not in the unsaved .gd file (it is in the saved version of the file).

![image](https://user-images.githubusercontent.com/22623259/147116685-5c2d2081-548a-4649-a924-cebe046106c4.png)
The crashed program when trying to Move to Method and the engine cannot find the method line reference.
}

When trying to access the "connected" method from the right tab, the engine tries to find the method in the file or opened file instance.
When the file is not saved, but there is an instance in which the method is found, the engine tries to jump to that method's line, but fails, because the opened file instance is shorter than the instance of the file, the method was found, that is why we get an out of bounds error.

If the method is not found in any instances, the program straight up crashes. I think it got in an infinite loop, but just guessing.

### Steps to reproduce

Setup file is in the Godot Engine Docs: https://docs.godotengine.org/en/stable/getting_started/step_by_step/scripting.html

When trying to add the first signal onto the Panel's sayhello.gd script from the connetion creation dialog, then deleting the connected method's code, the tab on the right does not update, so the "connected" signal is still selectable.

Unfortunately, I do not know, how to exactly reproduce it, because I get some side-bugs on the way, like duplicated .gd file instances and etc., which is totally undefined behaviour, but cannot get the program crashing again.
